lib/test/unit/test-suite-runner.rb

```diff
@@ -0,0 +1,105 @@
+#--
+#
+# Author:: Nathaniel Talbott.
+# Copyright:: Copyright (c) 2000-2003 Nathaniel Talbott. All rights reserved.
+# Copyright:: Copyright (c) 2008-2011 Kouhei Sutou. All rights reserved.
+# Copyright:: Copyright (c) 2024 Tsutomu Katsube. All rights reserved.
+# License:: Ruby license.
+
+module Test
+  module Unit
+    class TestSuiteRunner
+      def initialize(test_suite)
+        @test_suite = test_suite
+      end
+
+      def run(result, &progress_block)
+        @start_time = Time.now
+        yield(TestSuite::STARTED, @test_suite.name)
+        yield(TestSuite::STARTED_OBJECT, @test_suite)
+        run_startup(result)
+        run_tests(result, &progress_block)
+      ensure
+        begin
+          run_shutdown(result)
+        ensure
+          @elapsed_time = Time.now - @start_time
```

Comment:
Ah, we need to set `@start_time` and `@elapsed_time` of `@test_suite`.
Because `TestSuite` has `#start_time` and `#elapsed_time`.

How about set them in `TestSuite#run`?

```ruby
class TestSuite
  def run(result, &progress_block)
    runner = TestSuiteRunner.new(self)
    runner.run(result) do |event, *args|
      case event
      when STARTED
         @start_time = Time.now
      when FINISHED
         @elapsed_time = Time.now - @start_time
      end
      yield(event, *args)
    end
  end
end
```

Reply:
Ah, thanks! I understood that by using block, it's unnecessary to define setters of `TestSuite`. Fix it 7da9af4.

I noticed that `@start_time` and `@elapsed_time` of `TestSuiteRunner` are not used, so I removed them in 23c0bd9. Is this change okay?

Reply:
Yes!